uberfire-m2repo-editor/uberfire-m2repo-editor-backend/src/main/java/org/guvnor/m2repo/backend/server/M2ServletContextListener.java

```diff
@@ -0,0 +1,151 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.guvnor.m2repo.backend.server;
+
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.DirectoryStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Enumeration;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+import javax.servlet.ServletContext;
+import javax.servlet.ServletContextEvent;
+import javax.servlet.ServletContextListener;
+import javax.servlet.annotation.WebListener;
+
+import org.appformer.maven.integration.Aether;
+import org.eclipse.aether.artifact.Artifact;
+import org.eclipse.aether.artifact.DefaultArtifact;
+import org.eclipse.aether.installation.InstallRequest;
+import org.eclipse.aether.installation.InstallResult;
+import org.eclipse.aether.installation.InstallationException;
+import org.guvnor.common.services.project.model.GAV;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * It reads all the jars present in the WEB-INF/lib
+ * to create a Map with entries of GAV and path of the dependency
+ * */
+@WebListener
+public class M2ServletContextListener implements ServletContextListener {
+
+    private final String JAR_EXT = ".jar";
+    private final String WEB_INF_FOLDER = "WEB-INF";
+    private final String LIB_FOLDER = "lib";
+    private final String JAR_ARTIFACT = "jar";
+    private final String GROUP_ID = "groupId";
+    private final String ARTIFACT_ID = "artifactId";
+    private final String VERSION = "version";
+
+    private Logger logger = LoggerFactory.getLogger(M2ServletContextListener.class);
+
+    public M2ServletContextListener(){}
+
+    @Override
+    public void contextInitialized(ServletContextEvent servletContextEvent) {
+        logger.info("M2ServletContextListener contextInitialized started");
+        ServletContext ctx = servletContextEvent.getServletContext();
+        String jarsPath = ctx.getRealPath(getJarsFolder());
+        long startTime = System.nanoTime();
+        int jarsDeployed = deployJarsFromWar(jarsPath);
+        long endTime = System.nanoTime();
+        long totalTime = TimeUnit.NANOSECONDS.toSeconds(endTime - startTime);
+        logger.info("M2ServletContextListener contextInitialized deployed {} jars in {} sec ", jarsDeployed, totalTime);
+    }
+
+    @Override
+    public void contextDestroyed(ServletContextEvent servletContextEvent) {}
+
+    private String getJarsFolder(){
+        StringBuilder sb = new StringBuilder();
+        sb.append(File.separator).append(WEB_INF_FOLDER).append(File.separator).append(LIB_FOLDER).append(File.separator);
```

Comment:
I'd prefer to have this as constant:
JARS_FOLDER: File.separator + WEB_INF_FOLDER + File.separator  + LIB_FOLDER + File.separator

Reply:
Ok